DataFormats/Common/interface/Wrapper.h

```diff
@@ -73,20 +73,20 @@ namespace edm {
     std::shared_ptr<soa::TableExaminerBase> tableExaminer_() const override;
 
   private:
-    bool present;
     T obj;
+    bool present;
```

Comment:
As discussed privately, I want to understand better the implications of this change (even if, in principle, I think this order is better).

Reply:
I understand the generic rule that ordering members in decreasing size decreases memory size by reducing padding. Is the motivation for this change just setting a good example by following that principle? Or is there some other reasoning?

In this specific case, there are only 2 members and this reordering moves the padding from between the members to the end of the struct. If I understand how padding works correctly, there is no memory saving.

I've also heard that putting the most used variable first can be faster (quite possibly by a negligible amount). That might motivate putting "present" first. I don't remember why it was done this way in the first place.

There is also the issue that when a new release reads older data (from before and after this change), we would be forcing some streamer format conversion which takes time... Although it ought to work and would not matter when reading new data with a new release (most of the time).

Reply:
The reason for the change was I discovered Ref's are broken for objects which can be placed on a char memory boundary instead of on at least a short boundary. 
The internals of the RefCore expect the objects to live on an even memory boundary so the lowest bit of the memory address can be used to encode extra info.

Reply:
Interesting !
Would it make sense to add a comment about this, and to declare the `Wrapper` as `alignas(2)` or `alignas(4)` ?

Reply:
It is an interesting idea, but I don't think it is necessary. Wrapper is inheriting from WrapperBase which defined virtual functions. Therefore the first memory address of Wrapper is the pointer to the virtual table so that already requires an alignment greater than 1. Moving `obj` to be the first member data guarantees that `obj` is on a memory boundary equivalent to Wrapper as a whole.